public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -129,13 +129,13 @@ export class QueryEditorRow extends PureComponent<Props, State> {
     }
 
     this.setState({
-      datasource,
+      datasource: (datasource as unknown) as DataSourceApi<TQuery>,
```

Comment:
Probably out of the scope for this PR, but it would be really nice to get some well defined/flexible interfaces for the datasource types to prevent casts like this